cache/network/download.go

```diff
@@ -85,18 +87,55 @@ func downloadWithClient(ctx context.Context, httpClient *retryablehttp.Client, p
 	return matchedKey, err
 }
 
-func downloadFile(ctx context.Context, client *http.Client, url string, dest string, maxConcurrency uint) error {
+func downloadFile(ctx context.Context, httpClient *retryablehttp.Client, url string, dest string, maxConcurrency uint, logger log.Logger) error {
+	env := os.Getenv("BITRISEIO_DEPENDENCY_CACHE_MAX_IDLE_CONNS_PER_HOST")
```

Comment:
We could pass the envs as params (similar to maxConcurrency), so envs are read from envRepository and not directly from os.Getenv.